pkg/odo/cli/create/namespace/namespace.go

```diff
@@ -105,7 +108,29 @@ func (nco *NamespaceCreateOptions) Run(ctx context.Context) (err error) {
 	if err != nil {
 		return err
 	}
-
+	if nco.waitFlag {
+		var timeOut = time.After(nco.clientset.PreferenceClient.GetTimeout())
```

Comment:
The default value for `nco.clientset.PreferenceClient.GetTimeout()` is currently [1 second](https://odo.dev/docs/overview/configure#preference-key-table). Isn't that too low?

Reply:
Also, to me, if the user sets `--wait`, it means that they purposely are willing to wait until they get confirmation that the namespace/project is returned correctly. So IMO, we should not even have a timeout.
The user would need to interrupt the process to cancel the wait.
I'm thinking for example a script using `odo create namespace --wait` before doing other things..

Reply:
For me, it's a matter of 10's of milliseconds. Having to break manually can be dangerous if executed from a script: is the user expecting the command will never return in case of problem? I would instead return an error if the namespace is not created after a *reasonable* time. As the timeout is configurable, I would keep as is, and users in heavy loaded systems could increase it.

Reply:
Yup, that makes sense. Personally, I found this `Timeout` preference confusing and not used consistently. I think every command with a `--wait` flag should also provide an additional `--timeout` flag to set a timeout only for this command, but that's a different story.
Alright, in this case, returning an error if the timeout is reached would indeed make more sense. The error should also clearly indicate how the user can adjust this timeout, I think.

cc @valaparthvi